File: April Leetcode Challenge/1046. Last Stone Weight.java
class Solution {
    public int lastStoneWeight(int[] stones) {
      int n=stones.length;
        PriorityQueue<Integer> pq = new PriorityQueue<>(n, Collections.reverseOrder());
        for(int i=0;i<n;i++)
        {
            pq.add(stones[i]);
        }
        
        while(pq.size()>1)
        {
            int y=pq.poll();
            int x=pq.poll();
            if(y==x)
            {
                continue;
            }
            else
            {
            pq.add(y-x);
            }
            
        }
        if(pq.size()==0)
        {
            return 0;
        }
        return pq.peek();
        
    }
}

File: April Leetcode Challenge/1046. Last Stone Weight.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        priority_queue<int>pq;
        for(int i=0;i<stones.size();i++)
        {
          pq.push(stones[i]);  
        }
        while(pq.size()>1)
        {
            int y=pq.top();
            pq.pop();
            int x=pq.top();
            pq.pop();
            if(x==y)
            {
                continue;
            }
            else
            {
                 pq.push(y-x);
            }
        }
        if(pq.size()==0)
        {
            return 0;
        }
        else
        {
            return pq.top();
        }
    }
};
